datalake/src/main/java/com/sequenceiq/datalake/service/sdx/StackRequestManifester.java

```diff
@@ -306,6 +309,24 @@ void setupInstanceVolumeEncryption(StackV4Request stackRequest, String cloudPlat
                     encryption.setType(EncryptionType.DEFAULT);
                 }
             });
+        } else if (CloudPlatform.AZURE.name().equals(environmentResponse.getCloudPlatform())) {
+            String diskEncryptionSetId = Optional.of(environmentResponse)
+                    .map(DetailedEnvironmentResponse::getAzure)
+                    .map(AzureEnvironmentParameters::getResourceEncryptionParameters)
+                    .map(AzureResourceEncryptionParameters::getDiskEncryptionSetId)
+                    .orElse(null);
+            if (diskEncryptionSetId != null) {
```

Comment:
if possible, please use `Optional<String>` instead of `String` and in `if` condition  please use `diskEncryptionSetId.isPeresent()` instead of `diskEncryptionSetId != null`

Reply:
It can surely be done. However, while setting it in line no. 327, `azure.getEncryption().setDiskEncryptionSetId(diskEncryptionSetId)` it has to be instantiated to string as `setDiskEncryptionSetId` doesn't expects `Optional<String>`.

Reply:
https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#get-- you can get the value